### PR TITLE
Split the verification that needs CommonParams and doesn't need it

### PR DESCRIPTION
**Pull request by sgkim126**
It's to ready the ChangeParams.
The current code assumes that the CommonParams are immutable.

### CODE REVIEW

**Comment by sgkim126** [comment]
It depends on https://github.com/CodeChain-io/codechain/pull/1545

File: rpc/src/v1/impls/chain.rs [hunk 1]
```diff
@@ -113,7 +114,7 @@ where
             .client
             .get_text(transaction_hash, block_id)
             .map_err(errors::transaction_state)?
-            .map(|text| Text::from_core(text, self.client.common_params(block_number).unwrap().network_id())))
+            .map(|text| Text::from_core(text, self.client.common_params(block_id).unwrap().network_id())))
```
Comment (majecty) [inline rpc/src/v1/impls/chain.rs]: Should we use parent's block common params?

File: rpc/src/v1/impls/chain.rs [hunk 2]
```diff
@@ -237,7 +238,7 @@ where
     }
 
     fn get_min_transaction_fee(&self, action_type: String, block_number: u64) -> Result<Option<u64>> {
-        if let Some(common_parameters) = self.client.common_params(Some(block_number)) {
+        if let Some(common_parameters) = self.client.common_params(block_number.into()) {
```
Comment (majecty) [inline rpc/src/v1/impls/chain.rs]: Should we use parent's block common params?
If the RPC's caller wants to fee to creates transactions in the next block, the current implementation is OK.
However, if the RPC's caller wants to check the block's min_fee to verify the block, we should use the previous block's common params.

File: rpc/src/v1/impls/chain.rs [hunk 3]
```diff
@@ -219,16 +220,16 @@ where
     fn get_block_by_number(&self, block_number: u64) -> Result<Option<Block>> {
         let id = BlockId::Number(block_number);
         Ok(self.client.block(&id).map(|block| {
-            Block::from_core(block.decode(), self.client.common_params(Some(block_number)).unwrap().network_id())
+            Block::from_core(block.decode(), self.client.common_params(block_number.into()).unwrap().network_id())
```
Comment (HoOngEe) [inline rpc/src/v1/impls/chain.rs]: We can use predefined variable `id` instead of `block_number.into()`

File: rpc/src/v1/impls/chain.rs [hunk 4]
```diff
@@ -219,16 +220,16 @@ where
...
     }
 
     fn get_block_by_hash(&self, block_hash: H256) -> Result<Option<Block>> {
         let id = BlockId::Hash(block_hash);
         Ok(self.client.block(&id).map(|block| {
             let block = block.decode();
-            let block_number = block.header.number();
-            Block::from_core(block, self.client.common_params(Some(block_number)).unwrap().network_id())
+            let block_hash = block.header.hash();
+            Block::from_core(block, self.client.common_params(block_hash.into()).unwrap().network_id())
```
Comment (HoOngEe) [inline rpc/src/v1/impls/chain.rs]: We can use predefined variable `id` instead of `block_hash.into()`

File: rpc/src/v1/impls/chain.rs (hunk 2 above)
Comment (sgkim126) [reply rpc/src/v1/impls/chain.rs]: I make it use the parameters of the parent block.

**Pull request merged**